### PR TITLE
[WIP] Replace toasts with snackbars

### LINKED ISSUE #2388: Replace toasts with snackbars where appropriate
**Summary:** 

App uses toasts at many places which can be replaced by Snackbars for better UI.

**Commons app version:** 

Latest master

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by ShridharGoel**
**Description (required)**

Fixes #2388 

What changes did you make and why?

Replaced toasts with snackbars.

**Tests performed (required)**

Tested betaDebug on Android 7.1.2 (API 25)

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java [hunk 1]
```diff
@@ -28,6 +28,14 @@ public static void showShortSnackbar(View view, int messageResourceId) {
...
+        if (view.getContext() == null) {
+            return;
+        }
+
+        ExecutorUtils.uiExecutor().execute(() -> Snackbar.make(view, message, Snackbar.LENGTH_LONG).show());
+    }
+
     public static void showLongToast(Context context, String text) {
         if (context == null) {
             return;
```
Comment (ujjwalagrawal17) [inline app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java]: @ShridharGoel You can remove showLongToast method if it isn't used in any other place.

Comment (ujjwalagrawal17) [reply app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java]: @ShridharGoel Can you update this PR accordingly so that we can merge this.

Comment (ShridharGoel) [reply app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java]: @ujjwalagrawal17 Yes, sure.

Comment (domdomegg) [reply app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java]: showLongToast is now used in other places so should not be removed in this PR

File: app/src/main/java/fr/free/nrw/commons/campaigns/CampaignView.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public void setCampaign(Campaign campaign) {
         view.setVisibility(View.GONE);
         ((MainActivity) getContext()).defaultKvStore
                 .putBoolean("displayCampaignsCardView", false);
-        ViewUtil.showLongToast(getContext(),
+        ViewUtil.showLongSnackbar(getRootView(),
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/campaigns/CampaignView.java]: The issue reported by @domdomegg seems like related with this line.

**Comment by neslihanturan** [comment]
Please let us know if you are no longer volunteer to finalize the issue, so that someone else can complete @ShridharGoel :)

**Comment by ShridharGoel** [comment]
Anyone can feel free to take up the issue.

**Pull request closed without merging**